### PR TITLE
Update server-communication.md

### PR DESCRIPTION
**Pull request by angelism**
Updated the page.

### CODE REVIEW

File: src/angular/guide/server-communication.md [hunk 1]
```diff
@@ -371,17 +348,16 @@ This simple app conveys that idea, albeit imperfectly, in the way it handles a `
...
 <div id="subscribe"></div>
 ### _HeroListComponent_ error handling  {#hero-list-component}
 
-Back in the `HeroListComponent`, you wrapped the call to
-`_heroService.getHeroes()` in a `try` clause. When an exception is
-caught, the `errorMessage` variable &mdash; which you've bound conditionally in the
-template &mdash; gets assigned to.
+In the `HeroListComponent`, the call to
+`_heroService.getHeroes()` is wrapped in a `try` clause. When an error is
+generated, the `errorMessage` variable is assigned the error as follows:
```
Comment (sarahmaddox) [inline src/angular/guide/server-communication.md]: Here's some wording that combines your re-write with some parts of the original:

```
In the `HeroListComponent`, you wrapped the call to
`_heroService.getHeroes()`in a `try` clause. You also conditionally bound the
`errorMessage` variable in the template. When an exception is
caught, the `errorMessage` variable is assigned a value as follows:
```

File: src/angular/guide/server-communication.md [hunk 2]
```diff
@@ -395,42 +371,41 @@ template &mdash; gets assigned to.
...
 ## Send data to the server  {#post}
 
-So far you've seen how to retrieve data from a remote location using an HTTP service.
-Now you'll add the ability to create new heroes and save them in the backend.
+This page describes the functionality to create new heroes and save them in the code.
 
-You'll write a method for the `HeroListComponent` to call, a `create()` method, that takes
-just the name of a new hero. It begins like this:
+Create a method for the `HeroListComponent` to call a `create()` method. This method takes
+the name of a new hero.The following code sample explains this scenario:
```
Comment (sarahmaddox) [inline src/angular/guide/server-communication.md]: Here's a combination of your improvements with some parts of the original:

```
Write a `create()` method for the `HeroListComponent` to call. The `create()` method takes
the name of a new hero. See the following sample:
```

File: src/angular/guide/server-communication.md [hunk 3]
```diff
@@ -395,42 +371,41 @@ template &mdash; gets assigned to.
...
 
 <?code-excerpt "lib/src/toh/hero_service.dart (create-sig)"?>
 ```
   Future<Hero> create(String name) async {
 ```
 
-To implement it, you must know the server's API for creating heroes.
+To implement this scenario, the user must know the  server's API for creating heroes.
```
Comment (sarahmaddox) [inline src/angular/guide/server-communication.md]: Change "the user" to "you". It's best to address the readers directly, to ensure they know this is something they must do or know.

File: src/angular/guide/server-communication.md [hunk 4]
```diff
@@ -504,22 +480,21 @@ Wikipedia offers a modern `CORS` API and a legacy `JSONP` search API.
...
 {%comment%}
 ==============================================================================
 ==============================================================================
 ==============================================================================
 
 //- block wikipedia-jsonp+ unused portion from TS
-Wikipedia offers a modern `CORS` API and a legacy `JSONP` search API. This example uses the latter.
-The Angular `Jsonp` service both extends the `BrowserClient` service for JSONP and restricts you to `GET` requests.
+Wikipedia offers a modern `CORS` API and a legacy `JSONP` search API. This example uses this API.
+The Angular `Jsonp` service extends the `BrowserClient` service for JSONP and also restricts the `GET` requests.
```
Comment (sarahmaddox) [inline src/angular/guide/server-communication.md]: Let's keep the original wording in the case of the above 2 lines. It's technically more correct.

File: src/angular/guide/server-communication.md [hunk 5]
```diff
@@ -532,33 +507,32 @@ in `app.module.ts`.
...
 <div class="l-sub-section" markdown="1">
-    The `JSONP` technique requires that you pass a callback function name to the server in the query string: `callback=JSONP_CALLBACK`.
+    The `JSONP` technique requires you to pass a callback function name to the server in the query string: `callback=JSONP_CALLBACK`.
     The server uses that name to build a JavaScript wrapper function in its response, which Angular ultimately calls to extract the data.
-    All of this happens under the hood.
-</div>
+    </div>
 
-If you're looking for articles with the word "Angular", you could construct the query string by hand and call `jsonp` like this:
+To search for articles with the term, "Angular", construct the query string called `jsonp` as follows:
```
Comment (sarahmaddox) [inline src/angular/guide/server-communication.md]: Changed "called" to "and call"

File: src/angular/guide/server-communication.md [hunk 6]
```diff
@@ -569,85 +543,79 @@ The component's `search(term)` method delegates to the `WikipediaService`, which
...
+  The [async pipe](pipes.html#async-pipe) has read-only components, which do not interact with the data.
 
   `HeroListComponent` can't use the pipe because `addHero()` pushes newly created heroes into the list.
 </div>
 
 ## A wasteful app  {#wasteful-app}
 
-The Wikipedia search makes too many calls to the server.
-It is inefficient and potentially expensive on mobile devices with limited data plans.
+The Wikipedia search makes multiple calls to the server.This method is inefficient and potentially expensive on mobile devices with limited data plans.
```
Comment (sarahmaddox) [inline src/angular/guide/server-communication.md]: Add a space before "This".

File: src/angular/guide/server-communication.md [hunk 7]
```diff
@@ -569,85 +543,79 @@ The component's `search(term)` method delegates to the `WikipediaService`, which
...
 
 ### 1. Wait for the user to stop typing
 
-Presently, the code calls the server after every keystroke.
-It should only make requests when the user *stops typing*.
-Here's how it will work after refactoring:
+The code calls the server after every keystroke.
+and sends a request to the server after the user stops typing.
+The following sample code shows how the refactoring affects the code:
```
Comment (sarahmaddox) [inline src/angular/guide/server-communication.md]: Let's keep the original words in these 3 lines, as they're technically more correct.

File: src/angular/guide/server-communication.md [hunk 8]
```diff
@@ -569,85 +543,79 @@ The component's `search(term)` method delegates to the `WikipediaService`, which
...
+and sends a request to the server after the user stops typing.
+The following sample code shows how the refactoring affects the code: 
 
 <img class="image-display" src="{% asset_path 'ng/devguide/server-communication/wiki-2.gif' %}" alt="Wikipedia search app (v.2)" width="250">
 
 ### 2. Search when the search term changes
 
-Suppose a user enters the word *angular* in the search box and pauses for a while.
-The app issues a search request for *angular*.
+IF a user enters the word *angular* in the search box and pauses for a while, the app issues a search request for *angular*.
```
Comment (sarahmaddox) [inline src/angular/guide/server-communication.md]: Capitalisation - please change "IF" to "If".

File: src/angular/guide/server-communication.md [hunk 9]
```diff
@@ -569,85 +543,79 @@ The component's `search(term)` method delegates to the `WikipediaService`, which
...
 
-Then the user backspaces over the last three letters, *lar*, and immediately re-types *lar* before pausing once more.
-The search term is still _angular_. The app shouldn't make another request.
+If the user deletes the last three letters, *lar*, and re-types *lar* before pausing again.
+The search term is still _angular_. Therore, the app should not make another request.
```
Comment (sarahmaddox) [inline src/angular/guide/server-communication.md]: Typo: "Therore"

File: src/angular/guide/server-communication.md [hunk 10]
```diff
@@ -569,85 +543,79 @@ The component's `search(term)` method delegates to the `WikipediaService`, which
...
+The app issues two search requests, one for *angular* and the other for *http*.
 
-Which response arrives first? It's unpredictable.
-When there are multiple requests in-flight, the app should present the responses
-in the original request order.
-In this example, the app must always display the results for the *http* search
-no matter which response arrives first.
+The user does not know which response was recorded first. When there are multiple requests in-flight, the app presents the responses in the original request order.
+In this example, the app displays the results for the *http* search
+irrespective of the order of the response.
```
Comment (sarahmaddox) [inline src/angular/guide/server-communication.md]: Let's keep the original wording in the above 5 lines, as it better reflects the division between the app and the user.

File: src/angular/guide/server-communication.md [hunk 11]
```diff
@@ -569,85 +543,79 @@ The component's `search(term)` method delegates to the `WikipediaService`, which
...
 
 <div id="more-observables"></div>
 ## More fun with Observables
 
-You could make changes to the `WikipediaService`, but for a better
-user experience, create a copy of the `WikiComponent` instead and make it smarter,
-with the help of some nifty Observable operators.
+A user can make changes to the `WikipediaService` service. However, for a better
+user experience, create a copy of the `WikiComponent`. This can be done through,
+Observable operators.
```
Comment (sarahmaddox) [inline src/angular/guide/server-communication.md]: Here's some wording that addresses the reader as the developer:

```
You can make changes to the `WikipediaService` service. However, for a better
user experience, create a copy of the `WikiComponent` and make it smarter
with some Observable operators.
```

File: src/angular/guide/server-communication.md [hunk 12]
```diff
@@ -569,85 +543,79 @@ The component's `search(term)` method delegates to the `WikipediaService`, which
...
 <code-tabs>
   <?code-pane "lib/src/wiki/wiki_smart_component.dart" linenums?>
   <?code-pane "lib/src/wiki_component.dart" linenums?>
 </code-tabs>
 
 While the templates are virtually identical,
-there's a lot more RxJS in the "smart" version,
-starting with `debounceTime`, `distinctUntilChanged`, and `switchMap` operators,
-imported as [described above](#rxjs-library).
+there are more RxJS such as, `debounceTime`, `distinctUntilChanged`, and `switchMap` operators in the new version.
```
Comment (sarahmaddox) [inline src/angular/guide/server-communication.md]: Change "are" to "is".

File: src/angular/guide/server-communication.md [hunk 13]
```diff
@@ -666,9 +634,8 @@ The `WikipediaService` returns a separate Observable of string arrays (`Observab
 The user could issue multiple requests before a slow server has had time to reply,
 which means a backlog of response Observables could arrive at the client, at any moment, in any order.
 
-The `switchMap` returns its own Observable that _combines_ all `WikipediaService` response Observables,
-re-arranges them in their original request order,
-and delivers to subscribers only the most recent search results.
+The `switchMap` returns Observable that _combines_ all `WikipediaService` responses. Observables,
+re-arrange them in their original request order, and delivers only the most recent search results to subscribers.
```
Comment (sarahmaddox) [inline src/angular/guide/server-communication.md]: Let's keep the original wording in the above 3 lines, as it's technically more correct.

**Comment by kwalrath** [comment]
@angelism are you going to be able to (1) sign the CLA and (2) make Sarah's suggested changes?

**Comment by angelism** [comment]
Hi,

I signed the CLA long back (shared the screenshot with Sarah) and shared the issues I was having while trying to edit the page.  I am unable to incorporate Sarah’s changes.

Regards,
Nikita

**Comment by kwalrath** [comment]
Oh no, sounds like you're having issues with your GitHub login! No worries, Sarah & I will make sure that your changes get in. Thanks very much for your contribution!

**Comment by angelism** [comment]
Thanks Kathy for the opportunity to contribute to the project.

Regards,
Nikita

**Comment by kwalrath** [comment]
@sarahmaddox I wasn't able to use the SignCLA tool to confirm that the CLA is signed. Can you confirm? (I suspect @angelism has another email address.)

**Comment by sarahmaddox** [comment]
@kwalrath I can confirm that @angelism sent me a screenshot, showing that the Google CLA tool says she has already signed the Google Individual CLA.

Does @angelism need to do any more? I've advised her by email that she could do one of the following, if she has 2 email addresses:
* either sign in to the browser using the other email address, and sign the CLA with that address,
* or add another email address to GitHub at https://github.com/settings/emails - I think this will work, though the first option is the most fail-safe.

Many thanks
Sarah

**Comment by kwalrath** [comment]
It'd be great if @angelism could fix the CLA situation using one of the solutions you proposed.

If that doesn't work out for some reason, could you send me her other email address (privately is fine)?

**Comment by kwalrath** [comment]
Confirmed that the alternate email address has a signed CLA, so we can ignore the `cla: no` label.

**Comment by kwalrath** [comment]
Closing in favor of its replacement, #1222.

**Pull request closed without merging**